### ISSUE TITLE
ui screen glitches

### ISSUE THREAD

**Issue by runlykhel** [goatfungus/NMSSaveEditor#839]
still getting the screen glitches showing multiple NMSS opening and overlapping, just as reported earlier.
just downloaded last update did not correct the problem.
I'm using the java that's recommended on NMSS down load page.

**Comment by Kokopelli-Bob** [comment]
I am having same issue, off and on, last few weeks, in the  last week it got way worse almost unusable program. As a note using an  i9 12900 and 3080ti  Windows computer.

**Comment by runlykhel** [comment]
Hear ya, i9-13900k, 32gb ram, RTX 3080 w/12gb vram here. NMSSE worked fine until 1.4…updates.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by statopre** [comment]
same problem here. ui starts glitching on first action: right click, scrolling and stuff overlaps and gets weird.

**Comment by woodstockfurygit** [comment]
Thought it was just me and my laptop being funky attached to 2 other screens...

**Comment by Kokopelli-Bob** [comment]
an update note, I tried a different Java, type program, jdk the program works, but with the same issues. I do run 3 monitors, I disabled two of them, same issue. Windows 11. I have changed the refresh rates, no effect. also the scale  on the 4k monitors from 100% up to 200# and no effect either. I also changed the res. to 1080 and no change. I am posting this as what troubleshooting I have done on my system. thanks

**Comment by januxnet** [comment]
I searched here for my particular issue but could not find it, but maybe it is the same case for yours.  I had to reinstall the vm where I had the save editor running and so downloaded the latest version of the editor, I had the previous one, then installed latest version of java -in linux it was package  java-latest-openjdk, this in Fedora Linux-  and the save editor started as usual, showing the latest path where the save is, but not letting me selecting nor changing anything, not even the different saves in first tab, nor open new file, no errors all silent.
So tried downgrading java to 1.8, etc etc, but nothing changed.
Until I discovered NMSSaveEditor.conf    I believe this files gets created when the program runs, so I deleted it and the save editor opened like clean, as the first time that ask to select file,  and all worked again

TLDR: try deleting NMSSaveEditor.conf if you updated the save editor and is acting weird, maybe this works for you too.

**Comment by januxnet** [comment]
well after reading all the comments again I believe that my issue had nothing to do your this reported, I had no glitches in the screens nor tabs, just that could not change anything in the program as described in my comment above this one.

**Comment by runlykhel** [comment]
> well after reading all the comments again I believe that my issue had nothing to do with the ones reported here, I had no glitches in the screens nor tabs, just that could not change anything in the program as described in my comment above this one.

Agreed not related to our problem.

**Comment by pitsch** [comment]
![2023-06-16 16_16_14-No Man's Sky Save Editor - 1 14 2 (INTERCEPTOR)](https://github.com/goatfungus/NMSSaveEditor/assets/3827235/2c56c1b3-3a86-4b49-98d2-8521cc7adccf)
tried 3 different saves, same effect. disabling directx12 in java settings (Dsun.java2d.d3d=false  -Dsun.java2d.noddraw=true) and adding an exclusion rule to nvidia control panel (RX3070) didnt help. replacing latest java 8 (1.8.0-371b11) with corretto (8.372.07.1) changing resolutions and switching off second panel didnt help either.

**Comment by statopre** [comment]
Save file doesn't matter. The glitches happen even without any save loaded when I click on view menu for example.

**Comment by goatfungus** [comment]
> Save file doesn't matter. The glitches happen even without any save loaded when I click on view menu for example.

I'm still trying to figure this one out, as it doesn't happen on my PC at all.
If this is happening without any save loaded, that might be something to do with the account data and might be easier to debug. Can you please send me your log file as well as a copy of your save files (in zip format)?

**Comment by goatfungus** [comment]
> ![2023-06-16 16_16_14-No Man's Sky Save Editor - 1 14 2 (INTERCEPTOR)](https://user-images.githubusercontent.com/3827235/246443524-2c56c1b3-3a86-4b49-98d2-8521cc7adccf.png) tried 3 different saves, same effect. disabling directx12 in java settings (Dsun.java2d.d3d=false -Dsun.java2d.noddraw=true) and adding an exclusion rule to nvidia control panel (RX3070) didnt help. replacing latest java 8 (1.8.0-371b11) with corretto (8.372.07.1) changing resolutions and switching off second monitor didnt help either.

Would you be able to try running with the option -Dsun.java2d.opengl=true and see what that does?

**Comment by pitsch** [comment]
enabling hardware acceleration did not fix it. UI glitches got reduced quite a bit, but did not fully disappear, after i uninstalled oracle java to replace it with AdpoptOpenJdk https://adoptium.net/de/temurin/releases/?version=11

**Comment by Face0ff007** [comment]
having the same issue

**Comment by jacozilla** [comment]
Have this issue on new rig - Win 11 with correct JRE 8 installed. However, on my older Win 10 machine with same JRE 8, and using same save file to edit, have zero issues at all.

I’ve tried deleting conf file as someone suggested but the duplicate screens and overlay glitching always happens on Win 11 machine but never on Win 10

-same JRE 8 and nms save files
-Win 10, 4790k + 2070 Super works fine
-Win 11, 13900k + 4090 has the issue, and I’ve tried several older versions - all same problem

tried nms editor 14.11, several 13.11, and even reverted back to oldest ver I had on Win10 machine - v11.1.2. They all work fine on Win 10, all have same glitch on Win 11

dumb question - but should I be using newer version of Java on Win 11? I know JRE 8 is what we’re supposed to use, but Win 11 support is telling me to use Java 16?

**Comment by jacozilla** [comment]
Tried using command line option as suggested, manually launched jar file with --> java -jar NMSSaveEditor.jar -Dsun.java2d.opengl=true

1. Used brand new singularity expedition save, so no issues of prior save file edits. Started expedition, found ship, jumped in/out to make save and quit.

2. Launched editor with command line option above, opened this fresh save and got these warnings (which I don't get if opening main save file which was last saved ~Oct 2022 w/ whatever was then current editor ver. I think 11.x). Basically, if I open a non-singularity version save file, get these warnings. If open old save file not touched since long before current version (~Oct 2022), then zero error msgs when opening.
WARNING: name mapping not found: yID
  WARNING: name mapping not found: X;?
  WARNING: name mapping not found: WmI
  WARNING: name mapping not found: :?x
  WARNING: name mapping not found: IsL
  WARNING: name mapping not found: MMA
  WARNING: name mapping not found: me6
  WARNING: name mapping not found: 4q6
  WARNING: name mapping not found: AeI
  WARNING: name mapping not found: L8=
  WARNING: name mapping not found: waR
  WARNING: name mapping not found: rER
  WARNING: name mapping not found: OHU

3. Same screen duplication glitch as before, may not happen for few seconds or longer, but usually when clicking on something like resize cargo inventory the screen glitch happens. So same result using standard bat file launch or this extra command line option

4. Better explanation of Win 10 vs 11 editor results - fresh save file I edited on Win 10 machine does not have this glitch issue - but once I have edited the save and try using it on Win 11 PC, get two distinctly different behavior -->
a) after I edited it - but NOT yet entered game using that save file, can not load into editor on Win 11 machine that has this glitch. It just hangs or waits forever trying to load and shows this error in console
Exception in thread "AWT-EventQueue-0" java.lang.RuntimeException: duplicate key: StackSizeGroup
        at nomanssave.eY.a(Unknown Source)
        at nomanssave.ff.a(Unknown Source)
        at nomanssave.ff.a(Unknown Source)
        at nomanssave.ff.a(Unknown Source)
        at nomanssave.ff.a(Unknown Source)
        at nomanssave.ff.b(Unknown Source)
        at nomanssave.ff.a(Unknown Source)
        at nomanssave.ff.a(Unknown Source)
        at nomanssave.ff.a(Unknown Source)
        at nomanssave.ff.a(Unknown Source)
        at nomanssave.fP.a(Unknown Source)
        at nomanssave.fL.M(Unknown Source)
        at nomanssave.Application.l(Unknown Source)
        at nomanssave.Application.e(Unknown Source)
        at nomanssave.Application.a(Unknown Source)
        at nomanssave.D.setSelectedItem(Unknown Source)
        at javax.swing.JComboBox.setSelectedItem(Unknown Source)
        at javax.swing.JComboBox.setSelectedIndex(Unknown Source)
        at javax.swing.plaf.basic.BasicComboPopup$Handler.mouseReleased(Unknown Source)
        at java.awt.AWTEventMulticaster.mouseReleased(Unknown Source)
        at java.awt.Component.processMouseEvent(Unknown Source)
        at javax.swing.JComponent.processMouseEvent(Unknown Source)
        at javax.swing.plaf.basic.BasicComboPopup$1.processMouseEvent(Unknown Source)
        at java.awt.Component.processEvent(Unknown Source)
        at java.awt.Container.processEvent(Unknown Source)
        at java.awt.Component.dispatchEventImpl(Unknown Source)
        at java.awt.Container.dispatchEventImpl(Unknown Source)
        at java.awt.Component.dispatchEvent(Unknown Source)
        at java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
        at java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)
        at java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
        at java.awt.Container.dispatchEventImpl(Unknown Source)
        at java.awt.Window.dispatchEventImpl(Unknown Source)
        at java.awt.Component.dispatchEvent(Unknown Source)
        at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
        at java.awt.EventQueue.access$500(Unknown Source)
        at java.awt.EventQueue$3.run(Unknown Source)
        at java.awt.EventQueue$3.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.awt.EventQueue$4.run(Unknown Source)
        at java.awt.EventQueue$4.run(Unknown Source)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
        at java.awt.EventQueue.dispatchEvent(Unknown Source)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
        at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
        at java.awt.EventDispatchThread.run(Unknown Source)
        Suppressed: nomanssave.eX: Missing null terminator
                at nomanssave.ff.close(Unknown Source)
                ... 42 more
b) However, the save file seemingly works b/c can use it to enter game, see all modifications made such as new cargo sizes, material items added to inventory, multitools I had imported from prior exported files, tech blueprints installed into tech slots all working, etc.

Once I have entered game and then re-saved in-game, I can now open that save using editor on Win 11 - with similar warnings as before with what seems any singularity version save (even one never touched by editor, brand new fresh start)
Reading save file...
  Slot: Slot 3 - CUSTOM - Singularity
  Filename: save6.hg
  WARNING: name mapping not found: yID
  WARNING: name mapping not found: X;?
  WARNING: name mapping not found: WmI
  WARNING: name mapping not found: :?x
  WARNING: name mapping not found: IsL
  WARNING: name mapping not found: MMA
  WARNING: name mapping not found: me6
  WARNING: name mapping not found: 4q6
  WARNING: name mapping not found: AeI
  WARNING: name mapping not found: L8=
  WARNING: name mapping not found: waR
  WARNING: name mapping not found: rER
  WARNING: name mapping not found: OHU
Finished.

So from workaround pov, I can just xfer my files to my Win10 machine to edit, then copy back over. But not sure why Win11 (or some aspect of new PC) has this glitch issue.

**Comment by Droid8Apple** [comment]
Not trying to necro-post but yeah this is still a thing. Going fullscreen used to fix it. Is there really still not any fixes/workarounds?

**Comment by js-ware** [comment]
it seems like it's related to mouse movements over the NMS-SaveFileEditor window.  If I start it and just use the keyboard to move  around the tabs and text fields I don't see any issues, but as soon as I move the mouse over the window it starts "glitching", like it's trying to place current element of the editor window where the mouse cursor is.

**Comment by Rossatron-UI** [comment]
For others that experience an issue with the window replicating itself over and over on itself, I had to close MSI Afterburner, once done worked flawlessly.

**Comment by Kokopelli-Bob** [comment]
Thanks I will give that a try..
________________________________

**Comment by Rosedafool** [comment]
> For others that experience an issue with the window replicating itself over and over on itself, I had to close MSI Afterburner (and the Riva Statistics Server), once done worked flawlessly.

how dose one do this?

**Comment by wapperjaw00-droid** [comment]
When I first used the editor ... I had the same issue. Screen jumpy and flickering. Had to edit really fast, then save ... etc. It was very annoying. But I eventually solved this by switching from 120 Hz to 60 Hz. No more flickering and for some odd reason I can now keep it at 120 Hz now ... the issue is gone ...

**Comment by Syafiq2357** [comment]
Shutting down RivaTuner Statistics Server works for me.
